### PR TITLE
Task/adq 119 datamart patient integration tets

### PR DESCRIPTION
**Pull request by bschofield-va**
Adds integration tests to support Datamart using a minimally invasive approach.
1. A local copy of the Mitre database is now captured as an H2 database in `data-query-tests`. This file can be recreated on demand with utility scripts.
2. ResourceVerifier now has knowledge of which resources are datamart aware and will automatically execute queries with and with out the Datamart feature toggle set
3. To support both `cdw-schemas` file based data and exported Mitre data, a `cdw-schema` index has been created to map Mitre IDs to `cdw-schema` records. This index may be recreated as new resources are added using the `rebuild-datamart-index.sh` tool.
4. The file-based CDW database will now automatically reload changed index files 😄

### CODE REVIEW

**Comment by bschofield-va** [comment]
This is ready.

**Comment by bschofield-va** [comment]
I lied... I'm going to add something that prevents datamart failures in non-local environments. I don't really trust the Datamart enough yet for me to let it hold up a build

**Comment by bschofield-va** [comment]
OK really ready now

File: data-query-tests/src/main/java/gov/va/api/health/dataquery/tests/SystemDefinitions.java [hunk 1]
```diff
@@ -118,12 +118,12 @@ private static TestIds localIds() {
...
-                .given("JOHN Q")
-                .name("VETERAN,JOHN")
-                .family("VETERAN")
+                .birthdate("1948-06-28")
+                .given("Conrad619")
+                .name("Olson653, Conrad619")
+                .family("Olson653")
                 .build())
-        .patient("185601V825290")
+        .patient("111222333V000999") // ("185601V825290")
```
Comment (jblefkove-va) [inline data-query-tests/src/main/java/gov/va/api/health/dataquery/tests/SystemDefinitions.java]: Cruft?

File: data-query-tests/src/test/java/gov/va/api/health/dataquery/tests/ResourceVerifier.java [hunk 2]
```diff
@@ -84,13 +104,61 @@ public static ResourceVerifier get() {
...
+   * toggled off, we'll allow Datamart failures anywhere but locally.
+   */
+  private boolean datamartFailuresEnabled() {
+    if (Environment.get() == Environment.LOCAL) {
+      return true;
+    }
+    if (isTrue(toBoolean(System.getProperty("datamart.failures.enabled")))) {
+      return true;
+    }
+    if (isFalse(toBoolean(System.getenv("DATAMART_FAILURES_ENABLED")))) {
```
Comment (jblefkove-va) [inline data-query-tests/src/test/java/gov/va/api/health/dataquery/tests/ResourceVerifier.java]: This seems backward

File: data-query/src/main/java/gov/va/api/health/dataquery/service/mranderson/client/RestMrAndersonClient.java [hunk 3]
```diff
@@ -17,6 +18,7 @@
 
 /** A rest implementation of the Mr. Anderson client. */
 @Component
+@Slf4j
```
Comment (jblefkove-va) [inline data-query/src/main/java/gov/va/api/health/dataquery/service/mranderson/client/RestMrAndersonClient.java]: Cruft

**Pull request merged**